src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
+
 public class ArrayList<T> implements List<T> {
+    private static final int OFFSET = 1;
+    private static final int SIZE_ADJUSTMENT = 1;
+    private static final int START_INDEX = 0;
+    private static final double ELEMENT_DATA_MULTIPLIER = 1.5;
+    private int listCapacity = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[listCapacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == listCapacity) {
+            grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < START_INDEX) {
+            checkIndex(index);
+        }
+        if (size == listCapacity) {
+            grow();
+        }
+        final int s = size;
+        System.arraycopy(elementData, index, elementData, index + OFFSET, s - index);
+        elementData[index] = value;
+        size = s + SIZE_ADJUSTMENT;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (size + list.size() >= listCapacity) {
+            grow();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            elementData[size] = list.get(i);
+            size += SIZE_ADJUSTMENT;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        final int temporaryValue = size;
+        final T oldValue = elementData[index];
+        index += 1;
+        System.arraycopy(elementData, index, elementData, index - OFFSET, temporaryValue - index);
+        size = temporaryValue - SIZE_ADJUSTMENT;
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        final int temporaryValue = size;
+        for (int i = 0; i < size; i++) {
+            if (elementData[i] == null || elementData[i].equals(element)) {
+                System.arraycopy(elementData, i + OFFSET, elementData, i, temporaryValue - i);
+                size = temporaryValue - SIZE_ADJUSTMENT;
+                return element;
+            }
+        }
+        throw new NoSuchElementException("No such element in list: " + element);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    @Override
+    public String toString() {
+        return Arrays.toString(Arrays.copyOf(elementData, size));
+    }
+
+    private void grow() {
+        int size = (int) (elementData.length * ELEMENT_DATA_MULTIPLIER);
```

Comment:
Use informative names for your variables and methods. The variable 'size' in the grow method is confusing since there is already an instance variable with the same name. Consider renaming it to something like 'newCapacity'.